datafusion/expr-common/src/signature.rs

```diff
@@ -123,24 +127,29 @@ pub enum TypeSignature {
     /// One or more arguments belonging to the [`TypeSignatureClass`], in order.
     ///
     /// For example, `Coercible(vec![logical_float64()])` accepts
-    /// arguments like `vec![DataType::Int32]` or `vec![DataType::Float32]`
+    /// arguments like `vec![Int32]` or `vec![Float32]`
     /// since i32 and f32 can be cast to f64
     ///
     /// For functions that take no arguments (e.g. `random()`) see [`TypeSignature::Nullary`].
     Coercible(Vec<TypeSignatureClass>),
-    /// One or more arguments that can be "compared"
+    /// One or more arguments cast to single, comparable type.
+    ///
+    /// Each argument will be coerced to a single type using the
+    /// coercion rules described in [`comparison_coercion_numeric`].
```

Comment:
yes, we have numeric currently, update doc when code is updated to be consistent

Reply:
I agree -- this code could be improved / named better. I looked at the implementation and there are quite a few differences between `comparison_coercion_numeric` and `comparison_coercion` that I didn't understand and thus didn't attempt to document

It would be great if someone could